farmOS 3.2.0 blog post (#27)
farmOS/farmOS-community-blog: posts/2024/farmOS-3.2.0/index.md

---
title: farmOS 3.2.0
date: 2024-04-10
author: Michael Stenta
slug: 2024/farmOS-3.2.0
categories:
  - newsletter
---

# farmOS 3.2.0

We are excited to announce the release of
[farmOS 3.2.0](https://github.com/farmOS/farmOS/releases/tag/3.2.0)!
This release includes a number of new features and improvements. Here are some
of the highlights.

## Comments on farmOS records

If you've ever wanted the ability to have discussions on individual records, now
you can. Three new modules are available, which add the ability to comment on
logs, assets, and plans, respectively. Previously, the only way to record these

> kinds of thoughts or conversations was in the notes field of the record, which
> isn't ideal for obvious reasons.

Reviewer: ```suggestion
kinds of thoughts or conversations in farmOS was via the notes field of the record, which
isn't ideal for obvious reasons - i.e. lack of enforced structure, attribution, etc.
```

> Comments appear at the bottom of each record, and each comment has its own

Reviewer: ```suggestion
Comments appear at the bottom of each record. Each comment has its own
```

> author and timestamp, which helps to keep track of who chimed in on the

Reviewer: ```suggestion
author and timestamp which helps to keep track of who chimed in on the
```

> conversation and when. It's also possible to reply to comments, so a
> conversation can fork into multiple areas of focus.

Reviewer: Nit: "branch" (vs "fork") feels a little more natural/fitting here...

```suggestion
conversation and when. It's also possible to reply to comments, so a
conversation can branch out into multiple areas of focus.
```

## Filter by log notes

> On the April 10th, 2024 farmOS
> [monthly call](https://farmos.org/community/monthly-call/), Nick from

Reviewer: ```suggestion
On the [April 10th, 2024 farmOS
monthly call](https://farmos.discourse.group/t/farmos-monthly-call-10-april-2024/1972), Nick from
```

IMO, the forum topic is the better link because it includes more detail about this instance of the call - participants, agenda, etc - and also includes the link about monthly calls in general.

open a pull request to contribute that change back to farmOS. Here is the pull
request we created together: https://github.com/farmOS/farmOS/pull/825

Later that day, it was merged into farmOS and included in the 3.2.0 release.
This may have been the fastest feature-request turnaround ever! And a good
learning opportunity! Thanks Nick!

## Logs are done by default

A very simple, but helpful, change is that logs are now marked as "done" by
default. So if you are out in the field and you just need to quickly record
something you did, you don't need to change them from "pending" to "done". We
discussed this as a community and decided that this is a more sensible default
for the majority of users. A potential next step will be to automatically
toggle the status based on whether or not the date is in the past or future,
which will make things even easier!

## Files and images on taxonomy terms

It is now possible to upload files and images to all types of taxonomy terms.
Previously, only the "Plant type" terms had a field for images, but many people
requested the ability to add files and images to other types as well. For

> example, attaching PDFs to material type, or photos to animal types.

Reviewer: ```suggestion
example, attaching PDFs to material type terms, or photos to animal type terms.
```
